### DISCUSSION TITLE
Cross Room Communication in Server

### DISCUSSION THREAD

**Discussion by Samson-114** [colyseus/colyseus#593] [Q&A] [answered]
Is there any way that two different room can communicate in the server side just like send and onMessage such that room B will do responding action when room A emit a event.

**Comment by endel** [comment, marked as answer]
Hi @Samson-114, you can achieve this through the presence API via [`this.presence.subscribe()`](https://docs.colyseus.io/server/presence/#subscribetopic-string-callback-function) and [`this.presence.publish()`](https://docs.colyseus.io/server/presence/#publishtopic-string-data-any)

If you know the `roomId` of the other room that you want to communicate, you can use [`remoteRoomCall`](https://docs.colyseus.io/server/matchmaker/?h=remoteroomcall#remoteroomcallroomid-method-args) instead.

**Reply by Samson-114** [reply]
@endel thank you for your answer. I really appreciate it.
